src/ripple/protocol/impl/SOTemplate.cpp

```diff
@@ -21,42 +21,46 @@
 
 namespace ripple {
 
-void SOTemplate::push_back (SOElement const& r)
+SOTemplate::SOTemplate (std::initializer_list<SOElement> uniqueFields,
+    std::initializer_list<SOElement> commonFields)
+: elements_ ()
+, indices_ (SField::getNumFields () + 1, -1) // Unmapped indices == -1
 {
-    // Ensure there is the enough space in the index mapping
-    // table for all possible fields.
-    //
-    if (mIndex.empty ())
-    {
-        // Unmapped indices will be set to -1
-        //
-        mIndex.resize (SField::getNumFields () + 1, -1);
-    }
+    // Add all SOElements.
+    elements_.reserve (uniqueFields.size() + commonFields.size());
+    elements_.assign (uniqueFields);
+    for (SOElement const& elem : commonFields)
+        elements_.push_back (elem);
```

Comment:
I'd `insert` instead of the loop. I.e.: `elements_.insert(elements_.end(), commonFields)`